File: Commons/Data/src/main/java/com/greentree/commons/data/resource/ResourceAction.java
package com.greentree.commons.data.resource;

import java.io.Serializable;

import com.greentree.commons.action.observable.RunObservable;

public interface ResourceAction extends Serializable {
	
	RunObservable getOnCreate();
	RunObservable getOnDelete();
	RunObservable getOnModify();
	
}

File: Commons/Data/src/main/java/com/greentree/commons/data/resource/ResourceAction.kt
package com.greentree.commons.data.resource

import com.greentree.commons.action.observable.RunObservable
import java.io.Serializable

interface ResourceAction : Serializable {

	val onCreate: RunObservable
	val onDelete: RunObservable
	val onModify: RunObservable

	object Null : ResourceAction {

		override val onCreate: RunObservable
			get() = RunObservable.NULL
		override val onDelete: RunObservable
			get() = RunObservable.NULL
		override val onModify: RunObservable
			get() = RunObservable.NULL
	}
}